src/hooks/useViewportRows.ts

```diff
@@ -94,20 +94,79 @@ export function useViewportRows<R>({
     }
   }, [expandedGroupIds, groupedRows, rawRows]);
 
+  const { totalRowHeight, getRowTop, getRowHeight, findRowIdx } = useMemo(() => {
+    if (typeof rowHeight === 'number') {
+      return {
+        totalRowHeight: rowHeight * rows.length,
+        getRowTop: (rowIdx: number) => rowIdx * rowHeight,
+        getRowHeight: () => rowHeight,
+        findRowIdx: (offset: number) => Math.floor(offset / rowHeight)
+      };
+    }
+
+    const rowPositions: ({ height: number; top: number })[] = [];
+    let totalRowHeight = 0;
+    // Calcule the height of all the rows upfront. This can cause performance issues
```

Comment:
I did not notice any performance issues even on a large grid where editing triggers re-calculation